### ISSUE TITLE
Generate AWS images in Fedora Infrastructure

### ISSUE THREAD

**Issue by thrix** [fedora-eln/eln#19]
People would like to test their Github pull requests against Fedora ELN. To do that we would need ELN images to be generated same way as Rawhide images under Fedora Infrastructure AWS account. 

For Fedora CI request see: https://pagure.io/fedora-ci/general/issue/185

**Comment by thrix** [comment]
@bookwar @contyk could sombody look at this? We now have ELN support in packit, we want to test against ELN, but this is blocking this effort

**Comment by jkonecny12** [comment]
Hello,

I work on Anaconda and we would like to add ELN to others already supported test environments in Packit on our PRs. Right now we are using Packit to run basic tests for F34 and master but for ELN we are using it just for builds. It would be great to also use tests there.

Also our other projects would benefit from that. Namely https://github.com/rhinstaller/python-simpleline and https://github.com/rhinstaller/initial-setup is using only Packit to run tests on PR so it would be great to test also against ELN to find out we won't broke the build before merge and release of the code.

Could you please support AWS images to enable this?

**Comment by contyk** [comment]
Probably @sgallagher might be the best person to drive this.

**Comment by sgallagher** [comment]
Yeah, this is definitely a reasonable request. Let me pull in @hanzz as well.

**Comment by thrix** [comment]
@hanzz @sgallagher hi, any update around this? can we expect these images to land sometime, in what timeframe? thanks

**Comment by sgallagher** [comment]
Sorry, this fell completely off my radar. Let's put this on the upcoming meeting agenda and find someone to drive this.

**Comment by thrix** [comment]
@sgallagher no worries, thanks for putting it on radar!

**Comment by thrix** [comment]
Ok, the fedora-infrastructure issue was closed :) So we are looking here for updates then.

**Comment by nirik** [comment]
Just to note why I closed the infra one: We have no one around that knows fedimg (The ancient python2 uploader we have now). But the cloud sig is planning on a new, nicer, better thing, so I figured it would be good to leverage that for this request rather than trying to get our old thing working. I am not sure what the timeline they have is however. 
CC: @davdunc

**Comment by AdamWill** [comment]
For the record, on the call I got the impression that ELN composes had no images; now I see they do have traditional ISO images, I guess this is about cloud images. Those are less important to openQA.

**Comment by sgallagher** [comment]
Right, these are specifically about creating and uploading images for AWS.

We already produce [install ISOs](https://odcs.fedoraproject.org/composes/production/latest-Fedora-ELN/compose/BaseOS/x86_64/iso/) as well as [qcow VM images and container base images](https://odcs.fedoraproject.org/composes/production/latest-Fedora-ELN/compose/BaseOS/x86_64/iso/).

**Comment by thrix** [comment]
@sgallagher cool, who can make them appear in Fedora AWS, we would like to avoid doing it ourselves

**Comment by sgallagher** [comment]
Honestly, I think that's been something the Cloud SIG has been doing for Fedora. I don't have any idea how it happens. Maybe I can defer to @dustymabe here.

**Comment by dustymabe** [comment]
Cloud uses fedimg to upload the AMIs. @davdunc has been trying to replace it for some time now and may be close to it. I'm honestly surprised fedimg is still working.

**Comment by thrix** [comment]
@davdunc hi, could we get those ELN images to AWS pls? Now we have to upload them ourselves ...

**Comment by thrix** [comment]
@davdunc kindly pinging :) we still have some users wanting this

**Comment by davdunc** [comment]
Will make this functional by Feb 23 2024

**Comment by jkonecny12** [comment]
Hi, I wonder is this already solved or still in pending?

**Comment by thrix** [comment]
@jkonecny12 we are looking

**Comment by thrix** [comment]
@davdunc I still do not see them in AWS

**Comment by yselkowitz** [comment]
@davdunc @thrix any progress here?

**Comment by yselkowitz** [comment]
@davdunc what is the current status of this?

**Comment by yselkowitz** [comment]
@tdawson is this something you are able to do?

**Comment by nirik** [comment]
This should need to happen in https://pagure.io/cloud-image-uploader now. This is the fedimg replacement.

Can someone file a issue there? Or a PR? Since the images are being produced and they are just like the fedora ones it should be pretty easy to add them I would think.

**Comment by yselkowitz** [comment]
@nirik thanks, filed: https://pagure.io/cloud-image-uploader/issue/46

**Comment by jeremycline** [comment]
I believe ELN images have been available in AWS and Azure for a couple months, you can find recent AMI IDs for [aarch64](https://apps.fedoraproject.org/datagrepper/v2/search?topic=org.fedoraproject.prod.fedora_image_uploader.published.v1.aws.eln.BaseOS.aarch64) and [x86_64](https://apps.fedoraproject.org/datagrepper/v2/search?topic=org.fedoraproject.prod.fedora_image_uploader.published.v1.aws.eln.BaseOS.x86_64) in the AMQP messages.

Similarly, they're in Azure in the community gallery under the `fedora-5e266ba4-2250-406d-adad-5d73860d958f` public gallery ID with the `Fedora-Cloud-ELN-Arm64` and `Fedora-Cloud-ELN-x64` image definition names.

**Comment by yselkowitz** [comment]
If I go to EC2 and search images for "ELN" then a bunch of snapshots show up under the Community category.  Therefore, this appears to be done already.
